src/mono/mono/mini/mini-generic-sharing.c

```diff
@@ -1188,8 +1190,22 @@ get_wrapper_shared_vtype (MonoType *t)
 		if (findex >= 16)
 			break;
 	}
-	if (findex > 5)
-		return NULL;
+
+	guint32 align;
+	int size = mono_class_value_size (klass, &align);
+
+	if (align == 4 && size <= 4 * 5) {
+		findex = size / align;
```

Comment:
So if I have a struct with 8 int8 fields and one int32 field, we're going to share that with a valuetuple_3_class ?

Reply:
This stuff is only used to share the gsharedvt in/out functions, which are used to transition in/out of gsharedvt code, those don't care about the contents of the valuetypes, just their size/layout.
Maybe we need to restrict this to int types or to wasm.

Reply:
ok, makes sense.

if it's just functions used for functions where we know what they're doing, I think it's fine to use this for all workloads.